Fix wrong filename and path name for matchers documentation (#60711)
saltstack/salt: doc/topics/matchers/index.rst

        return bool(self.opts["id"] in tgt)

> Place this code in a file called ``list_match.py`` in a directory ``matchers`` in your
> ``extension_modules`` directory. This path is defined in the master configuration.
> Sync this down to your minions with :py:func:`saltutil.sync_matchers <salt.modules.saltutil.sync_matchers>`.

Reviewer: ```suggestion
Place this code in a file called ``list_match.py`` in a ``_matchers`` directory in your
``file_roots``. Sync this down to your minions with 
:py:func:`saltutil.sync_matchers <salt.modules.saltutil.sync_matchers>`.
```
